Failing tests in DocStrings

Checking the Python files in NLTK with "python -m doctest" reveals that many tests are failing. In many cases, the failures are just cosmetic discrepancies between the expected and the actual output, such as missing a blank line, or unescaped linebreaks. Other cases may be real bugs.

If these failures could be avoided, it would become possible to improve CI by running "python -m doctest" each time a python file was modified.

Here is an updated list, after applying #2990, and unzipping those corpora marked unzip="1":

> nltk/parse/corenlp.py: 9 passed and 33 failed.
nltk/parse/stanford.py: 1 passed and 16 failed.
nltk/corpus/reader/framenet.py: 79 passed and 9 failed.
nltk/tokenize/regexp.py: 14 passed and 7 failed.
nltk/draw/table.py: 0 passed and 6 failed.
nltk/tokenize/simple.py: 7 passed and 6 failed.
nltk/text.py: 16 passed and 5 failed.
nltk/tokenize/stanford_segmenter.py: 3 passed and 5 failed.
nltk/draw/util.py: 1 passed and 4 failed.
nltk/tag/hunpos.py: 1 passed and 4 failed.
nltk/tokenize/__init__.py: 6 passed and 4 failed.
nltk/translate/ibm1.py: 12 passed and 4 failed.
nltk/tag/brill_trainer.py: 23 passed and 3 failed.
nltk/tokenize/destructive.py: 8 passed and 3 failed.
nltk/tokenize/treebank.py: 32 passed and 3 failed.
nltk/tokenize/util.py: 27 passed and 3 failed.
nltk/translate/meteor_score.py: 8 passed and 3 failed.
nltk/corpus/reader/categorized_sents.py: 5 passed and 2 failed.
nltk/corpus/reader/opinion_lexicon.py: 3 passed and 2 failed.
nltk/corpus/reader/reviews.py: 8 passed and 2 failed.
nltk/metrics/agreement.py: 8 passed and 2 failed.
nltk/parse/dependencygraph.py: 13 passed and 2 failed.
nltk/parse/evaluate.py: 6 passed and 2 failed.
nltk/parse/util.py: 5 passed and 2 failed.
nltk/tag/stanford.py: 2 passed and 2 failed.
nltk/tokenize/stanford.py: 3 passed and 2 failed.
nltk/corpus/__init__.py: 1 passed and 1 failed.
nltk/corpus/reader/comparative_sents.py: 5 passed and 1 failed.
nltk/corpus/reader/__init__.py: 1 passed and 1 failed.
nltk/corpus/reader/pros_cons.py: 2 passed and 1 failed.
nltk/probability.py: 39 passed and 1 failed.
nltk/sentiment/util.py: 7 passed and 1 failed.
nltk/stem/arlstem2.py: 3 passed and 1 failed.
nltk/stem/rslp.py: 3 passed and 1 failed.
nltk/stem/snowball.py: 6 passed and 1 failed.
nltk/tag/__init__.py: 12 passed and 1 failed.
nltk/tokenize/casual.py: 9 passed and 1 failed.
nltk/translate/ibm2.py: 18 passed and 1 failed.
nltk/translate/ibm3.py: 23 passed and 1 failed.
nltk/translate/ibm4.py: 25 passed and 1 failed.
nltk/translate/ibm5.py: 22 passed and 1 failed.

Good work!

I've experienced issues 1 through 3 that you listed. These make some sense, as these services simply require some additional setup. That said, the 4th one is odd. That might be indicative of some actual bugs.

As for the deprecated errors, we'll end up wanting to include the doctests in the CI check, so we should probably ensure that all tests pass. In that case, we might want to either remove the tests for the deprecated work, or modify them such that they're skipped. (I prefer the latter).

I would hope to use `pytest --doctest-modules .\nltk\` to run the doctests easily, but I'm running into issues there, likely because of our LazyLoaders:
```
...\lib\doctest.py:939: in find
    self._find(tests, obj, name, module, source_lines, globs, {})
.env39\lib\site-packages\_pytest\doctest.py:522: in _find
    doctest.DocTestFinder._find(  # type: ignore
...\lib\doctest.py:995: in _find
    for valname, val in obj.__dict__.items():
E   RuntimeError: dictionary changed size during iteration
```

Are you aware of a way to easily run the doctests perhaps?

--- a/nltk/corpus/reader/wordnet.py
+++ b/nltk/corpus/reader/wordnet.py
@@ -1178,7 +1178,8 @@ class WordNetCorpusReader(CorpusReader):
             warnings.warn(
                 "The multilingual functions are not available with this Wordnet version"
             )
-            self.omw_langs = set()
+
+        self.omw_langs = set()
 
         # A cache to store the wordnet data of multiple languages
         self._lang_data = defaultdict(list)

--- a/nltk/corpus/reader/wordnet.py
+++ b/nltk/corpus/reader/wordnet.py
@@ -1279,9 +1279,8 @@ class WordNetCorpusReader(CorpusReader):
         if lang in self._lang_data.keys():
             return
 
-        if self.langs() == ["eng"]:
-            self.add_provs(self._omw_reader)
-            self.omw_langs = set(self.provenances.keys())
+        if self._omw_reader and not self.omw_langs:
+            self.add_omw()
 
         if lang not in self.langs():
             raise WordNetError("Language is not supported.")
@@ -1309,11 +1308,15 @@ class WordNetCorpusReader(CorpusReader):
             file_name, file_extension = os.path.splitext(langfile)
             if file_extension == ".tab":
                 lang = file_name.split("-")[-1]
-                if lang in self.provenances.keys():
+                if lang in self.provenances.keys() or prov in ["cldr", "wikt"]:
                     # We already have another resource for this lang,
                     # so we need to further specify the lang id:
                     lang = f"{lang}_{prov}"
                 self.provenances[lang] = prov
+
+    def add_omw(self):
+        self.add_provs(self._omw_reader)
+        self.omw_langs = set(self.provenances.keys())
 
     def add_exomw(self):
         """
@@ -1327,6 +1330,7 @@ class WordNetCorpusReader(CorpusReader):
         """
         from nltk.corpus import extended_omw
 
+        self.add_omw()
         self._exomw_reader = extended_omw
         self.add_provs(self._exomw_reader)
 

